### ISSUE TITLE
Ability to auto log out and or quit after all missions are sent or partially sent

### ISSUE THREAD

**Issue by theruan** [TLDRMissions/TLDRMissions#618]
TLDR version:  10.1-013

Hi, i was trying to make a weakauras of this idea, but if u could implement in the addon, that would be very nice.

The idea is to be able to either logout or quit the game once all missions are sent(that can be sent).
IF weekly mission is delivered or picked( and not completed).

I often put missions to calculate at my job and have to let the PC running with other stuff along, having the client running uses a lot of cpu time and slows my other tasks, if it could quit or at least logout it would be nice.

If that can't be done, how do we call the calculate function from a weakauras?(without using the /click button)

**Comment by teelolws** [comment]
For your final question, this should do it, though I can't login right now to check:

`TLDRMissionsFrameCalculateButton:Click()`

Unfortunately, I cannot make the player logout or exit the game, and you won't be able to make a weakaura do that either. Both functions are only usable by Blizzard code:
https://warcraft.wiki.gg/wiki/API_Logout
https://warcraft.wiki.gg/wiki/API_Quit

**Comment by theruan** [comment]
If I understood it correctly, the protected functions can be called by a keypress, so in theory(limited knowledge on lua, so if i'm crazy, my apologies):
A 2 key presses do all(Assuming Space is bound):
Hit Space starts the completion and processing missions
HIt Space again starts a function using C_Timer.After which loops in itself every few seconds and once all missions are sent runs the logout or quit.
I know weakauras can use /camp /logout for instance:
__
leftclick = "/logout"

local region = WeakAuras.regions[aura_env.id].region
local frame = CreateFrame("Button","paddingHelperButton", region, "SecureActionButtonTemplate")
frame:SetAllPoints(region)
frame:SetAttribute("type1","macro")
frame:SetAttribute("macrotext1", leftclick)
frame:RegisterForClicks("AnyDown")
___
What I have no clue is how to iterate C_Timer.After in a secure format

**Comment by teelolws** [comment]
We can't call hardware event restricted functions or buttons such as /logout from within a C_Timer, and I won't be able to "click" a button registered to /logout when the addon finishes its calculations.
